### PR TITLE
dirs: use alt root when checking classic confinement support without …

### PR DESCRIPTION
**Pull request by bboozzoo**
…symlink workaround

The test needs be isolated from local filesystem state.

### CODE REVIEW

File: dirs/dirs_test.go [hunk 1]
```diff
@@ -93,7 +97,15 @@ func (s *DirsTestSuite) TestClassicConfinementSupportOnSpecificDistributions(c *
 	} {
 		reset := release.MockReleaseInfo(&release.OS{ID: t.ID, IDLike: t.IDLike})
 		defer reset()
-		dirs.SetRootDir("/")
+
+		// make a new root directory each time to isolate the test from
+		// local filesystem state and any previous test runs
+		altRoot, err := ioutil.TempDir("",
```
Comment (zyga) [inline dirs/dirs_test.go]: `dirs.SetRoot(c.MkDir())`

File: dirs/dirs_test.go [hunk 2]
```diff
@@ -93,7 +97,15 @@ func (s *DirsTestSuite) TestClassicConfinementSupportOnSpecificDistributions(c *
...
+			"TestClassicConfinementSupportOnSpecificDistributions")
+		c.Assert(err, IsNil)
+		defer os.RemoveAll(altRoot)
```
Comment (zyga) [inline dirs/dirs_test.go]: No need to remove when using `c.MkDir`

File: dirs/dirs_test.go [hunk 3]
```diff
@@ -100,12 +99,7 @@ func (s *DirsTestSuite) TestClassicConfinementSupportOnSpecificDistributions(c *
 
 		// make a new root directory each time to isolate the test from
 		// local filesystem state and any previous test runs
-		altRoot, err := ioutil.TempDir("",
-			"TestClassicConfinementSupportOnSpecificDistributions")
-		c.Assert(err, IsNil)
-		defer os.RemoveAll(altRoot)
-
-		dirs.SetRootDir(altRoot)
+		dirs.SetRootDir(c.MkDir())
```
Comment (zyga) [inline dirs/dirs_test.go]: Note that this is "persistent" so the suite or the test needs to undo that.

Comment (bboozzoo) [reply dirs/dirs_test.go]: There's a `defer dirs.SetRootDir("/")` at the beginning of this function.

**Pull request merged**